R/make_vaccineff_data.R

```diff
@@ -203,40 +203,77 @@ summary.vaccineff_data <- function(object, warnings_log = FALSE, ...) {
       checkmate::test_class(object, "vaccineff_data")
   )
 
-  cat(paste0("\nCohort start: ", object$start_cohort))
-  cat(paste0("\nCohort end: ", object$end_cohort, "\n"))
-
-  cat(object$warnings_log$filtered, sep = "")
-
-  # Extract tags
   tags <- linelist::tags(object$cohort_data)
 
-  # Summary
-  if (!is.null(object$matching)) {
-    cat("\nNearest neighbors matching iteratively performed.\n")
-    summary.match(object$matching)
+  summ <- list(
+    start_cohort = object$start_cohort,
+    end_cohort = object$end_cohort,
+    tags = tags,
+    trunc_log = object$warnings_log$filtered
+  )
 
-    if (warnings_log && !is.null(object$matching)) {
-      cat("\nWarnings:\n")
-      cat(object$warnings_log$matching, sep = "")
+  if (!is.null(object$matching)) {
+    summ$summary_vaccination <- object$matching$summary
+    summ$balance_all <- object$matching$balance_all
+    summ$balance_match <- object$matching$balance_match
+    summ$iterations <- object$matching$iterations
+    if (warnings_log) {
+      summ$match_log <- object$warnings_log$matching
+    } else {
+      summ$match_log <- NULL
     }
-
   } else {
-    summ_cohort <- match_summary(
+    summ_vacc <- match_summary(
       all = object$cohort_data,
       matched = NULL,
       vacc_status_col = tags$vacc_status_col
     )
-    cat("\nNo matching routine invoked.\n")
-    print(summ_cohort)
+    summ$summary_vaccination <- summ_vacc
+  }
+
+  class(summ) <- "summary.vaccineff_data"
```

Comment:
I would recommend changing this class name to use an underscore separator between words. As `summary()` is an S3 generic function in R, then `summary.<class>()` is a class-specific summary method and this may be confusing for others reading or using the code. The underscore (`summary_vaccineff_data`) will alleviate this.

Reply:
solved in  062d6a2b55cd2dc08d2c435871cab32d6c9144d1